File: src/programs/java/Factorial.java
package programs.java;

public class Factorial {

    public static void main(String args[]) {
        System.out.print(factorial(4));
    }

    static int factorial(int value) {
        if (value <= 1)
            return value;
        return value * factorial(value-1);
    }

}

File: src/programs/kotlin/Factorial.kt
package programs.kotlin

class Factorial {

    companion object{

        @JvmStatic
        fun main(args:Array<String>){
            print(factorial(4))
        }

        private fun factorial(value: Int): Int {
            if (value <= 1) return value
            return value * factorial(value - 1)
        }
    }


}